src/spaces/utilities.jl

```diff
@@ -55,6 +55,26 @@ function edistance(p1::ValidPos, p2::ValidPos, model::ABM{<:GridSpace{D,true}})
     sqrt(total)
 end
 
+"""
+    get_direction(from, to, model::ABM)
+Returns the direction vector from `from` to `to` taking into account periodicity of the space
+"""
+get_direction(from, to, model::ABM) = get_direction(from, to, model.space)
+
+function get_direction(from::NTuple{D,Float64}, to::NTuple{D,Float64}, space::ContinuousSpace{D,true}) where {D}
+    all_dirs = [to .+ space.extent .* (i, j) .- from for i in -1:1, j in -1:1]
+    return all_dirs[argmin(map(x -> sum(x .^ 2), all_dirs))]
+end
+
+function get_direction(from::NTuple{D,Int64}, to::NTuple{D,Int64}, space::GridSpace{D,true}) where {D}
+    all_dirs = [to .+ size(space.s) .* (i, j) .- from for i in -1:1, j in -1:1]
+    return all_dirs[argmin(map(x -> sum(x .^ 2), all_dirs))]
+end
+
+function get_direction(from, to, ::Union{GridSpace,ContinuousSpace})
+    return to .- from
+end
```

Comment:
This seems like a rather inefficient implementation. Why do you have to make so many vectors? You make a vector with `all_dirs`, you make another vector with `map`. Two alternatives:

1. Loop over possibilities and simply keep track of minimum
2. Just use the `round` function to get the best/proper direction instead of `argmin` of all directions. This doesn't really make sense.

Also, the way this is written, it explicitly assumes 2D-space, yet any dimensional space can be given as an input.

Reply:
Thanks for pointing this out! It looks like I had 2D space in mind when making this and didn't generalize it. I'll rework this function